SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -0,0 +1,880 @@
+          direct_iteration[iZone][iInst] = new CFluidIteration(config_container[iZone]);
+          direct_output[iZone] = new CFlowIncOutput(config_container[iZone], nDim);
+          break;
+        case DISC_ADJ_HEAT:
+          direct_iteration[iZone][iInst] = new CHeatIteration(config_container[iZone]);
+          direct_output[iZone] = new CHeatOutput(config_container[iZone], nDim);
+          break;
+        case DISC_ADJ_FEM:
+          SU2_MPI::Error("There is no multizone discrete adjoint functionality for problems including elasticity yet.",
+                         CURRENT_FUNCTION);
+          direct_iteration[iZone][iInst] = new CFEAIteration(config_container[iZone]);
+          direct_output[iZone] = new CElasticityOutput(config_container[iZone], nDim);
+          break;
+
+        default:
+          SU2_MPI::Error("There is no discrete adjoint functionality for one of the specified solvers yet.",
+                         CURRENT_FUNCTION);
+      }
+    }
+
+    direct_output[iZone]->PreprocessHistoryOutput(config_container[iZone], false);
+  }
+}
+
+CDiscAdjMultizoneDriver::~CDiscAdjMultizoneDriver(){
+
+  for (iZone = 0; iZone < nZone; iZone++){
+    for (iInst = 0; iInst < direct_nInst[iZone]; iInst++){
+      delete direct_iteration[iZone][iInst];
+    }
+    delete [] direct_iteration[iZone];
+  }
+
+  delete[] direct_iteration;
+  delete[] direct_nInst;
+
+}
+
+void CDiscAdjMultizoneDriver::StartSolver() {
+
+  /*--- Main external loop of the solver. Runs for the number of time steps required. ---*/
+
+  if (rank == MASTER_NODE)
+    cout << endl <<"------------------------------ Begin Solver -----------------------------" << endl;
+
+  if (rank == MASTER_NODE){
+    cout << endl << "Simulation Run using the Discrete Adjoint Multizone Driver" << endl;
+
+    if (driver_config->GetTime_Domain())
+      SU2_MPI::Error("The discrete adjoint multizone driver is not ready for unsteady computations yet.",
+                     CURRENT_FUNCTION);
+  }
+
+  for (iZone = 0; iZone < nZone; iZone++){
+
+    /*--- Set the value of the external iteration to TimeIter. -------------------------------------*/
+    /*--- TODO: This should be generalised for an homogeneous criteria throughout the code. --------*/
+    config_container[iZone]->SetTimeIter(0);
+    
+  }
+  
+  /*--- We directly start the (steady-state) discrete adjoint computation. ---*/
+
+  Run();
+  
+  /*--- Output the solution in files. ---*/
+
+  Output(TimeIter);
+
+}
+
+void CDiscAdjMultizoneDriver::Run() {
+
+  bool            checkSensitivity  = false;
+  unsigned short  jZone             = 0,
+                  wrt_sol_freq      = config_container[ZONE_0]->GetVolume_Wrt_Freq();
+  unsigned long   iIter             = 0,
+                  nIter             = 0,
+                  iOuter_Iter       = 0,
+                  iInner_Iter       = 0;
+
+  if (nZone == 1) {
+    nIter = driver_config->GetnIter();
```

Comment:
So since yesterday, the discrete adjoint multizone driver will be launched whenever we have a driver/master config file with `SOLVER= MULTIPHYSICS` and `MATH_PROBLEM = DISCRETE_ADJOINT`, regardless of the number of zones. Which totally makes sense.

Also the marked lines above are removed as they are also not needed anymore. `CDiscAdjMultizoneDriver` will run the specified number of outer iteration in the outer loop (...) :)

I run quite some tests to be sure that both (single zone driver and multizone driver for nZone=1) give same adjoints and sensitivities.